ui/src/app/apps/apps.component.ts

```diff
@@ -35,12 +42,21 @@ export class AppsComponent implements OnInit {
     private router: Router ) {
     }
 
+  /**
+   * As soon as the page loads we retrieve a list
+   * of {@link AppRegistration}s.
+   */
   ngOnInit() {
     this.loadAppRegistrations(false);
   }
 
+  /**
+   * Load a paginated list of {@link AppRegistration}s.
+   *
+   * @param reload
+   */
   public loadAppRegistrations(reload: boolean) {
-    this.busy = this.appsService.getApps(reload).subscribe(
+    this.busy.push(this.appsService.getApps(reload).subscribe(
```

Comment:
While you stay on apps page that array just keeps growing when you go on different pages, while moving between tabs will re-init a class. I'd prefer not to leave this type of code here as possible copy/pasting will then introduce it to somewhere else. At minimum we should create an issue to track this if we merge without a fix.